shared/eventbus/eventbus.go

```diff
@@ -75,32 +92,47 @@ func (b *BrokerService) Register(topic, description string, typ any) error {
 }
 
 func (b *BrokerService) EventTypes() []EventType {
+	// there's really no race condition here, but we need this lock to pass race condition tests.
+	b.eventMutex.Lock()
 	types := make([]EventType, 0, len(b.eventTypes))
 	for _, event := range b.eventTypes {
 		types = append(types, event)
 	}
+	b.eventMutex.Unlock()
+
 	sort.Slice(types, func(i, j int) bool {
 		return types[i].Topic < types[j].Topic
 	})
 	return types
 }
 
 func (b *BrokerService) Publish(topic string, ctx request.CTX, data any) error {
-	// there's really no race condition here so no mutex required
+	// no need for mutex since no race condition here
+	// this is done not to receive events after Stop() was called
+	// Since here we have not global mutex(for performance reasons) "couple of"
+	// events might pass through even after calling Stop(), but it should be Ok.
+	if b.stopping {
```

Comment:
I was about to comment on this, and verified that this is also caught by the race detector. Could we use an atomic value?

Reply:
There are no atomic operations on bool in Go 1.18, so I used int32 instead.